_source/_posts/2021-12-03-angular-dynamic-components.md

```diff
@@ -0,0 +1,773 @@
+
+{% img blog/angular-dynamic-components/overview.gif alt:"animated image showing logging in and displaying dynamic contents" width:"800" %}{: .center-image }
+
+In this post, we'll
+* Set up an Angular application with routing and lazy-loaded modules
+* Create an OIDC client in Okta
+* Add authentication to the Angular application
+* Capture authenticated user information and user claims
+* Simulate an external server call and implement the dynamic components
+* Complete displaying the dynamic components using the user claim value
+
+We'll be covering a lot in this post to build out a non-trivial Angular application. As a result, this post assumes some Angular knowledge. We'll be blazing through some parts to spend more time working through the dynamic components and user profile information.
+
+If you are new to Angular, check out the following fantastic resources and guides first
+* [Angular Quickstart](https://angular.io/guide/what-is-angular)
+* [A Quick Guide to Angular and GraphQL](/blog/2021/10/22/angular-graphql#integrate-oidc-for-auth)
+
+**Prerequisites**
+* A valid version of [Node](https://nodejs.org/) for Angular v13, such as Node 14.15
+* [Angular CLI](https://cli.angular.io)
+* [Okta CLI](https://cli.okta.com/)
+
+{% include toc.md %}
+
+## Scaffold the Angular application
+
+The [Angular CLI](https://cli.angular.io) automates creating Angular apps quickly. It makes the project folder and scaffolds the application shell. We'll pass in the parameters to
+* add routing
+* use scss for styles
+* use inline templates
+* use inline styles
+* skip tests (the code repo includes tests to show working tests with testbed setup and authentication service spies, so feel free to include tests if you want to try that yourself)
+
+We'll have a lot of small components in this application, so having inline templates and styles allow us to minimize the number of files to touch.
+
+Run the following command to create an Angular v13 app.
+
+```bash
+npx @angular/cli@13 new dynamic-components --routing --style=scss --inline-template --inline-style --skip-tests
+```
+
+You now have a directory named `dynamic-components` with a working application skeleton. All CLI commands in the rest of this post should be run inside the project directory.
+
+Next, we'll add the Angular Material component library to the application using their schematic. We'll pass in the parameters to
+* set the theme
+* add typography
+* enable animations
+
+Run the following command with preset values to add the Angular Material v13 library and affirm that you will proceed with the installation.
+
+```bash
+ ng add @angular/material@13 --theme=custom --typography=true --animations=true
+```
+
+## Create components for the initial view
```

Comment:
May be a a callout asking user to run the app with ng run or non start at this point might be nice, so that further changes will live reload as the progress